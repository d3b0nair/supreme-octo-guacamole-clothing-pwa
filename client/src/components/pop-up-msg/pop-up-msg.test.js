import React from "react";
import {
  render,
  fireEvent,
  waitFor,
  screen,
  cleanup,
} from "@testing-library/react";
import PopUpMsg from "./pop-up-msg.component";

describe("PopUpMsg component", () => {
  const mockOnDismiss = jest.fn();
  const mockAction = jest.fn();
  const mockProps = (otherProps) => ({
    ...otherProps,
    msg: "",
    onDismiss: mockOnDismiss,
    action: mockAction,
    actionText: "",
  });
  const component = (mockProps) => <PopUpMsg {...mockProps} />;

  afterEach(cleanup);

  it("should mount", () => {
    const { container } = render(component(mockProps()));
    expect(container).toMatchSnapshot();
  });
  it("should call onDismiss when CloseBtn clicked", async () => {
    render(component(mockProps()));
    await waitFor(() => fireEvent.click(screen.getByTestId("CloseBtn")));
    expect(mockOnDismiss).toHaveBeenCalled();
  });
  it("should call action when CustomButton clicked", async () => {
    render(component(mockProps()));
    await waitFor(() => fireEvent.click(screen.getByTestId("CustomButton")));
    expect(mockAction).toHaveBeenCalled();
  });
});
