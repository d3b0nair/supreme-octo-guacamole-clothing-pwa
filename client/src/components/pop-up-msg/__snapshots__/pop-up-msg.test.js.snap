// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopUpMsg component should mount 1`] = `
<div>
  <div
    class="sc-gsDKAQ kVZzFU"
  >
    <span
      class="sc-hKwDye dLUxtG"
      data-testid="CloseBtn"
    >
      <svg>
        closeBtn.svg
      </svg>
    </span>
    <p
      class="sc-dkPtRN diaXdX"
    />
    <button
      class="sc-bdvvtL gRKetO"
      data-testid="CustomButton"
      style="width: 100%;"
    />
  </div>
</div>
`;
